src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -592,7 +592,19 @@ public Solution WithProjectMetadataReferences(ProjectId projectId, IEnumerable<M
         /// specified analyzer reference.
         /// </summary>
         public Solution AddAnalyzerReference(ProjectId projectId, AnalyzerReference analyzerReference)
-            => AddAnalyzerReferences(projectId, ImmutableArray.Create(analyzerReference));
+        {
+            if (projectId == null)
```

Comment:
Isn't nullalbe enabled here?

Reply:
Yes, but this is public API surface area.